chore: editorial review (#111)
KILTprotocol/docs: docs/concepts/04_credentials/02_ctypes.md

<!-- TODO: Replace with dynamically-generated JSON -->
```js title="CType schema example"
{
  $id: 'kilt:ctype:0xda3861a45e0197f3ca145c2c209f9126e5053fas503e459af4255cf8011d5101',
  $schema: 'http://kilt-protocol.org/draft-01/ctype#',
  title: 'CtypeModel 2',
  properties: {
    name: { type: 'string' },
    age: { $ref: `kilt:ctype:0x0147d10ef72a1b0c4a8e0147d10ef72a1b0c4a8e0147d10ef72a1b0c4a8e82af` },
    },
  type: 'object'
}
```

The CType schema is afterwards wrapped into the full CType object:

<!-- TODO: Replace with dynamically-generated JSON -->
```js title="Full CType example"
{
  schema:  {...} // as defined in the example above,
  owner: null, // DID of the owner, or null
  hash: '0xda3861a45e0197f3ca145c2c209f9126e5053fas503e459af4255cf8011d5101' // For looking up on-chain. It matches the `$id` property defined above
}
```

## CType Metadata

CType Metadata can be linked to a given CType to provide title and descriptions in different languages for the whole CType and its properties.

<!-- TODO: Add example of CType metadata -->

## Hashing

The hash of the CType is used to identify and anchor it to the KILT blockchain.

### Constructing the `hash` for the `$id`

> KILT uses the hashing algorithm `blake2b` to compute the hash of CTypes.

Reviewer: I think we should clarify that we use blake2b256 since the default variant of black2b has 512 bit output. Blake2s has a default of 256 bit, so this was a bit confusing for me in the beginning.

Before hasing, the CType object is sorted by a canonicalization algorithm to ensure that semantically equivalent CTypes with different order of their properties result in the same final hash.

The hash is computed from the following fields of the CType schema:

- `$schema`
- `properties`
  - `key`
  - `$ref`
  - `type`
  - `format`
- `title`
- `type`

The `$id` property, if present, is excluded from the hashing process since it represents the result of such a process, and is overwritten with the resulting hash, prepended with `kilt:ctype:` to form a URI.
Hence, a typical CType ID would look like this: `kilt:ctype:0xda3861a45e0197f3ca145c2c209f9126e5053fas503e459af4255cf8011d5101`.

## Storing and Querying CTypes

CTypes can be stored on the blockchain.

> After creating a CType, its full content can be included in the blockchain block history, while its hash is anchored to the blockchain state.

Reviewer: ```suggestion
After creating a CType, its full content is only included in the blockchain block history, while its hash is anchored to the blockchain state.
```
